central/declarativeconfig/manager_impl.go

```diff
@@ -60,21 +74,27 @@ var protoTypesOrder = []reflect.Type{
 // New creates a new instance of Manager.
 // Note that it will not watch the declarative configuration directories when created, only after
 // ReconcileDeclarativeConfigurations has been called.
-func New(reconciliationTickerDuration, watchIntervalDuration time.Duration, updaters map[reflect.Type]updater.ResourceUpdater, reconciliationErrorReporter ReconciliationErrorReporter) Manager {
+func New(reconciliationTickerDuration, watchIntervalDuration time.Duration, updaters map[reflect.Type]updater.ResourceUpdater,
+	reconciliationErrorReporter integrationhealth.Reporter, nameExtractor types.NameExtractor, idExtractor types.IDExtractor) Manager {
 	writeDeclarativeRoleCtx := declarativeconfig.WithModifyDeclarativeResource(context.Background())
 	writeDeclarativeRoleCtx = sac.WithGlobalAccessScopeChecker(writeDeclarativeRoleCtx,
 		sac.AllowFixedScopes(
 			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
 			// TODO: ROX-14398 Replace Role with Access
 			sac.ResourceScopeKeys(resources.Role, resources.Access)))
 	return &managerImpl{
-		universalTransformer:         transform.New(),
-		transformedMessagesByHandler: map[string]protoMessagesByType{},
-		reconciliationTickerDuration: reconciliationTickerDuration,
-		watchIntervalDuration:        watchIntervalDuration,
-		updaters:                     updaters,
-		reconciliationCtx:            writeDeclarativeRoleCtx,
-		reconciliationErrorReporter:  reconciliationErrorReporter,
+		universalTransformer:           transform.New(),
+		transformedMessagesByHandler:   map[string]protoMessagesByType{},
+		reconciliationTickerDuration:   reconciliationTickerDuration,
+		watchIntervalDuration:          watchIntervalDuration,
+		updaters:                       updaters,
+		reconciliationCtx:              writeDeclarativeRoleCtx,
+		declarativeConfigErrorReporter: reconciliationErrorReporter,
+		errorsPerDeclarativeConfig:     map[string]int32{},
+		idExtractor:                    idExtractor,
+		nameExtractor:                  nameExtractor,
+		shortCircuitSignal:             concurrency.NewSignal(),
```

Comment:
nit: is it necessary to initialize signals? Don't think we did that before

Reply:
See the description:
>the release tests for ^^ didn't fail because the signals (both stopSignal & shortCircuitSignal) were not initialized via concurrency.NewSignal(), hence stopSignal.IsDone() always yielded true

Currently, the reconciliation didn't work because we always stopped the reconciliation loop.